app/app.go

```diff
@@ -227,29 +237,33 @@ func wireP2P(ctx context.Context, life *lifecycle.Manager, conf Config, manifest
 }
 
 // wireCoreWorkflow wires the core workflow components.
-func wireCoreWorkflow(ctx context.Context, life *lifecycle.Manager, conf Config, manifest Manifest, nodeIdx NodeIdx, tcpNode host.Host) error {
+func wireCoreWorkflow(ctx context.Context, life *lifecycle.Manager, conf Config,
+	lock cluster.Lock, nodeIdx cluster.NodeIdx, tcpNode host.Host,
+) error {
 	// Convert and prep public keys and public shares
 	var (
 		corePubkeys    []core.PubKey
 		pubkeys        []eth2p0.BLSPubKey
 		pubshares      []eth2p0.BLSPubKey
 		pubSharesByKey = make(map[*bls_sig.PublicKey]*bls_sig.PublicKey)
-		threshold      int
 	)
-	for _, dv := range manifest.DVs {
-		threshold = dv.Threshold()
+	for _, dv := range lock.Validators {
+		pubkey, err := dv.PublicKey()
+		if err != nil {
+			return err
+		}
 
-		corePubkey, err := tblsconv.KeyToCore(dv.PublicKey())
+		corePubkey, err := tblsconv.KeyToCore(pubkey)
 		if err != nil {
 			return err
 		}
 
-		pubkey, err := tblsconv.KeyToETH2(dv.PublicKey())
+		pk, err := tblsconv.KeyToETH2(pubkey)
```

Comment:
should colocate this `pk`. code may be confusing because there are two pubkeys: `pubkey` and `pk`.

```go
pk, err := tblsconv.KeyToETH2(pubkey)
if err != nil {
   return err
}

pubkeys = append(pubkeys, pk)
```

Reply:
think about what makes them different and name them with longer identifiers that represent that